test/XRP/xrp-client-integration.test.ts

```diff
@@ -318,4 +318,75 @@ describe('XrpClient Integration Tests', function (): void {
     )
     assert.deepEqual(transactionStatus, TransactionStatus.Succeeded)
   })
+
+  it('Unauthorize Sending Account - failure on unauthorizing self', async function (): Promise<
+    void
+  > {
+    this.timeout(timeoutMs)
+    // GIVEN an existing testnet account
+    // WHEN unauthorizeSendingAccount is called with the account's own address
+    const result = await xrpClient.unauthorizeSendingAccount(
+      wallet.getAddress(),
+      wallet,
+    )
+
+    // THEN the transaction fails due to a malformed transaction.
+    const transactionHash = result.hash
+    const transactionStatus = result.status
+
+    assert.exists(transactionHash)
+    assert.equal(transactionStatus, TransactionStatus.MalformedTransaction)
+  })
+
+  it('Unauthorize Sending Account - failure on unauthorizing account that is not authorized', async function (): Promise<
+    void
+  > {
+    this.timeout(timeoutMs)
+    // GIVEN an existing testnet account that has not authorized any accounts
+    await xrpClient.enableDepositAuth(wallet)
+
+    // WHEN unauthorizeSendingAccount is called on an account that is not authorized
+    const sendingWallet = await XRPTestUtils.randomWalletFromFaucet()
+    const result = await xrpClient.unauthorizeSendingAccount(
+      sendingWallet.getAddress(),
+      wallet,
+    )
+
+    // THEN the transaction fails and the cost of the transaction is claimed by the network.
+    const transactionHash = result.hash
+    const transactionStatus = result.status
+
+    assert.exists(transactionHash)
+    assert.equal(transactionStatus, TransactionStatus.ClaimedCostOnly)
+  })
+
+  it('Unauthorize Sending Account - cannot send funds after an authorized account is unauthorized', async function (): Promise<
+    void
+  > {
+    this.timeout(timeoutMs)
+    // GIVEN an existing testnet account that has authorized another account to send XRP to it
+    const sendingWallet = await XRPTestUtils.randomWalletFromFaucet()
+
+    await xrpClient.enableDepositAuth(wallet)
+
+    await xrpClient.authorizeSendingAccount(sendingWallet.getAddress(), wallet)
```

Comment:
should we verify that a send works here before unauthorizing?

Reply:
if it's cool with you i'll probably leave it as is - `'Authorize Sending Account - can send funds after authorize'` tests that sending works after auth and `'Unauthorize Sending Account - failure on unauthorizing account that is not authorized'` tests that unauthorize doesn't work on an account that hasn't been authorized first